### PR TITLE
allow test utils be extended

### PR DESCRIPTION
**Pull request by camilamacedo86**
**Description**
Allow the test utils to be used for other projects as well.

### CODE REVIEW

File: test/e2e/utils/test_context.go [hunk 1]
```diff
@@ -43,11 +43,12 @@ type KBTestContext struct {
 	Resources  string
 	ImageName  string
 	Kubectl    *Kubectl
+	BinName    string
 }
 
 // TestContext init with a random suffix for test KBTestContext stuff,
 // to avoid conflict when running tests synchronously.
-func TestContext(env ...string) (*KBTestContext, error) {
+func TestContext(binName string, env ...string) (*KBTestContext, error) {
```
Comment (estroz) [inline test/e2e/utils/test_context.go:51]:
```suggestion
func NewTestContext(binName string, env ...string) (*KBTestContext, error) {
```

Probably a good idea to generalize `KBTestContext` to `TestContext` too.

File: test/e2e/utils/util.go [hunk 2]
```diff
@@ -24,6 +24,10 @@ import (
 	"strings"
 )
 
+const (
+	KubeBinName = "kubebuilder"
```
Comment (estroz) [inline test/e2e/utils/util.go:28]:
```suggestion
	KubebuilderBinName = "kubebuilder"
```

**Comment by estroz** [comment]
These modifications to e2e utils will be beneficial to any plugin developers, since they expose wrappers for `init`, `create api`, and `create webhook`.

/cc @DirectXMan12 @droot @mengqiy

File: pkg/test/test_context.go [hunk 3]
```diff
@@ -32,8 +34,9 @@ const prometheusOperatorVersion = "0.33"
 const certmanagerURL = "https://github.com/jetstack/cert-manager/releases/download/%s/cert-manager.yaml"
 const prometheusOperatorURL = "https://raw.githubusercontent.com/coreos/prometheus-operator/release-%s/bundle.yaml"
 
-// KBTestContext specified to run e2e tests
-type KBTestContext struct {
+//nolint:golint
+// TestContext specified to run e2e tests\
+type TestContext struct {
```
Comment (camilamacedo86) [inline pkg/test/test_context.go:39]: TestContext is not a good name because it will be at the end test.TestContext. @estroz have you a better idea?
PS.: Context can not be used here.  (redeclared issue because of ginkgo import)

**Comment by camilamacedo86** [comment]
/hold

Until get OK from 2 reviewers.

**Comment by camilamacedo86** [comment]
c/c @joelanford

File: pkg/test/utils/test_context.go [hunk 4]
```diff
@@ -32,8 +32,8 @@ const prometheusOperatorVersion = "0.33"
 const certmanagerURL = "https://github.com/jetstack/cert-manager/releases/download/%s/cert-manager.yaml"
 const prometheusOperatorURL = "https://raw.githubusercontent.com/coreos/prometheus-operator/release-%s/bundle.yaml"
 
-// KBTestContext specified to run e2e tests
-type KBTestContext struct {
+// TestContext specified to run e2e tests\
```
Comment (hasbro17) [inline pkg/test/utils/test_context.go:35]: Nit: Typo on the trailing `\`
```suggestion
// TestContext specified to run e2e tests
```

File: pkg/test/utils/test_context.go [hunk 5]
```diff
@@ -43,12 +43,13 @@ type KBTestContext struct {
 	Resources  string
 	ImageName  string
 	Kubectl    *Kubectl
+	BinName    string
```
Comment (hasbro17) [inline pkg/test/utils/test_context.go:46]: Can we please use the full var name here like we're doing for other fields? e.g `ImageName` or `TestSuffix`.
That's more readable imo.
```suggestion
	BinaryName    string
```

File: pkg/test/utils/test_context.go [hunk 6]
```diff
@@ -43,12 +43,13 @@ type KBTestContext struct {
...
 }
 
-// TestContext init with a random suffix for test KBTestContext stuff,
+// NewTestContext init with a random suffix for test TestContext stuff,
 // to avoid conflict when running tests synchronously.
-func TestContext(env ...string) (*KBTestContext, error) {
-	testSuffix, err := randomSuffix()
+func NewTestContext(binName string, env ...string) (*TestContext, error) {
```
Comment (hasbro17) [inline pkg/test/utils/test_context.go:51]:
```suggestion
func NewTestContext(binaryName string, env ...string) (*TestContext, error) {
```

**Comment by camilamacedo86** [comment]
/hold cancel
(received review of 2 persons and suggestions are addressed already)

File: pkg/test/utils/test_context.go [hunk 7]
```diff
@@ -32,8 +32,8 @@ const prometheusOperatorVersion = "0.33"
 const certmanagerURL = "https://github.com/jetstack/cert-manager/releases/download/%s/cert-manager.yaml"
```
Comment (estroz) [inline pkg/test/utils/test_context.go:32]: I'm not sure that we want to move these utils into `pkg`, since that indicates that this is a supported library. Can we leave these in `test/e2e` for now and make the library decision later?

File: testdata/project-v2-addon/go.mod [hunk 8]
```diff
@@ -9,5 +9,5 @@ require (
 	k8s.io/apimachinery v0.18.2
 	k8s.io/client-go v0.18.2
 	sigs.k8s.io/controller-runtime v0.6.0
-	sigs.k8s.io/kubebuilder-declarative-pattern v0.0.0-20200512162422-ce639cbf6d4c
+	sigs.k8s.io/kubebuilder-declarative-pattern v0.0.0-20200522144838-848d48e5b073
```
Comment (mengqiy) [inline testdata/project-v2-addon/go.mod:12]: Is this change really needed in this PR?

File: testdata/project-v3-addon/go.mod [hunk 9]
```diff
@@ -9,5 +9,5 @@ require (
 	k8s.io/apimachinery v0.18.2
 	k8s.io/client-go v0.18.2
 	sigs.k8s.io/controller-runtime v0.6.0
-	sigs.k8s.io/kubebuilder-declarative-pattern v0.0.0-20200512162422-ce639cbf6d4c
+	sigs.k8s.io/kubebuilder-declarative-pattern v0.0.0-20200522144838-848d48e5b073
```
Comment (mengqiy) [inline testdata/project-v3-addon/go.mod:12]: Same here

File: testdata/project-v2-addon/go.mod (hunk 8 above)
Comment (camilamacedo86) [reply testdata/project-v2-addon/go.mod:12]: solved here: https://github.com/kubernetes-sigs/kubebuilder/pull/1532 - It is related to the scenario that we need to  solve over it.

File: testdata/project-v3-addon/go.mod (hunk 9 above)
Comment (camilamacedo86) [reply testdata/project-v3-addon/go.mod:12]: ditto.

**Comment by camilamacedo86** [comment]
The nit is solved. So, all ok to /lgtm this one?
